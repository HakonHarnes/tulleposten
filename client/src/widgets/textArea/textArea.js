// @flow
import * as React from 'react';
import { Component } from 'react-simplified';

import style from './css/text-area.module.css';

class TextAreaLight extends Component<{
   onChange: any => mixed,
   className: string,
   placeholder?: string,
   value?: string,
   required: boolean
}> {
   render() {
      return (
         <textarea
            data-testid="text-area-light"
            className={`${this.props.className} ${style['text-area']} ${style['text-area-light']}`}
            placeholder={this.props.placeholder}
            value={this.props.value}
            onChange={this.props.onChange}
            required={this.props.required}
         />
      );
   }
}

class TextAreaDark extends Component<{
   onChange: any => mixed,
   className: string,
   placeholder?: string,
   value?: string,
   required: boolean
}> {
   render() {
      return (
         <textarea
            data-testid="text-area-dark"
            className={`${this.props.className} ${style['text-area']} ${style['text-area-dark']}`}
            placeholder={this.props.placeholder}
            value={this.props.value}
            onChange={this.props.onChange}
            required={this.props.required}
         />
      );
   }
}

export class TextArea {
   static Light = TextAreaLight;
   static Dark = TextAreaDark;
}
