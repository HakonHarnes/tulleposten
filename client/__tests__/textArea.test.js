//@flow
import * as React from 'react';
import ReactDOM from 'react-dom';

import '@testing-library/jest-dom/extend-expect';

import { render, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { TextArea } from '../src/widgets/textArea/textArea';

const changeHandler = jest.fn();

afterEach(cleanup);

it('renders without crashing', () => {
   const div = document.createElement('div');
   ReactDOM.render(<TextArea.Light onChange={changeHandler} className="area" required={false} />, div);
   ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
   const { getByTestId } = render(<TextArea.Light onChange={changeHandler} value="Vetle" className="area" required={true} />);
   expect(getByTestId('text-area-light')).toHaveValue('Vetle');
   expect(getByTestId('text-area-light')).toBeRequired();
   expect(getByTestId('text-area-light')).toHaveClass('area');
});

it('matches snapshot', () => {
   const tree = renderer.create(<TextArea.Light onChange={changeHandler} value="Vetle" className="area" required={false} />).toJSON();
   expect(tree).toMatchSnapshot();
});
