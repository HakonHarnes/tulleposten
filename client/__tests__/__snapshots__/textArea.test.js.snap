// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<textarea
  className="area text-area text-area-light"
  data-testid="text-area-light"
  onChange={[MockFunction]}
  required={false}
  value="Vetle"
/>
`;
